### ISSUE TITLE
Dependency Track Chart 1.0 Ingress Error on service format

### ISSUE THREAD

**Issue by smercier74** [evryfs/helm-charts#57]
**Describe the bug**
When we install the dependency track 4.1.0 with the chart 1.0.0 with an ingress gateway enabled to expose the backend and frontend services. During the helm chart installation i get a validation error:
```
error validating "": error validating data: [ValidationError(Ingress.spec.rules[0].http.paths[0].backend): unknown field "service" in io.k8s.api.networking.v1beta1.IngressBackend, ValidationError(Ingress.spec.rules[0].http.paths[1].backend): unknown field "service" in io.k8s.api.networking.v1beta1.IngressBackend]
helm.go:81: [debug] error validating "": error validating data: [ValidationError(Ingress.spec.rules[0].http.paths[0].backend): unknown field "service" in io.k8s.api.networking.v1beta1.IngressBackend, ValidationError(Ingress.spec.rules[0].http.paths[1].backend): unknown field "service" in io.k8s.api.networking.v1beta1.IngressBackend]
```

**Version of Helm and Kubernetes**:
Helm 3.5.1
EKS Cluster: 1.18.9

**Which chart**:
Dependency Track 4.1.0 / Chart 1.0

**What happened**:
There is a validation error during the ingress services (frontend/backend) analysis. Maybe an error linked with the version of K8s apis used.

**What you expected to happen**:
The chart can be used with Kubernetes 1.18

**How to reproduce it** (as minimally and precisely as possible):
Install the dependency track with the chart and the config below for the ingress:
```
# Ingress entry point
ingress:
  enabled: true
  host: <Deptrak_FQDN>
  annotations: 
    kubernetes.io/ingress.class: nginx
    kubernetes.io/tls-acme: "true"
    ingress.kubernetes.io/affinity: cookie
    ingress.kubernetes.io/ssl-redirect: "true"
    cert-manager.io/cluster-issuer: letsencrypt-issuer
  tls:
    enabled: true
    secretName: "dependency-track-tls"
```

**Anything else we need to know**:
I updated the code below:
```
  rules:
  - host: {{ .Values.ingress.host }}
    http:
      paths:
      - path: /api
        pathType: Prefix
        backend:
          service:
            name: {{ include "common.names.fullname" . }}-apiserver
            port:
              number: {{ .Values.apiserver.service.port }}
      - path: /
        pathType: Prefix
        backend:
          service:
            name: {{ include "common.names.fullname" . }}-frontend
            port:
              number: {{ .Values.frontend.service.port }}
```
Like that and it works fine after
```
  rules:
  - host: {{ .Values.ingress.host }}
    http:
      paths:
      - path: /api
        pathType: Prefix
        backend:
          serviceName: {{ include "common.names.fullname" . }}-apiserver
          servicePort: {{ .Values.apiserver.service.port }}
      - path: /
        pathType: Prefix
        backend:
          serviceName: {{ include "common.names.fullname" . }}-frontend
          servicePort: {{ .Values.frontend.service.port }}
```

**Comment by davidkarlsen** [comment]
Thanks for reporting, can you use this helper https://github.com/bitnami/charts/blob/master/bitnami/common/templates/_ingress.tpl and then I think it will work on various ingress-api versions.

**Comment by smercier74** [comment]
I looked the code of the helper and i think it could be good to use it in the ingress.yaml. However i'm not familiar with helm helpers and functions.  
Is it possible to give me an example how to use the function "common.ingress.backend" with dynamic values:  
-  {{ include "common.names.fullname" . }}-apiserver   
-  {{ .Values.apiserver.service.port }}   

Thanks in advance

**Comment by davidkarlsen** [comment]
@smercier74 See https://github.com/bitnami/charts/tree/master/bitnami/common#ingress
Can you open a PR please?

**Comment by davidkarlsen** [comment]
Fixed by #70
